File: Ahorn/entities/CustomCoverupWall.jl
module VivHelperCustomCoverupWall
using ..Ahorn, Maple
using Ahorn.VivHelper

@mapdef Entity "VivHelper/CustomCoverupWall" CustomCoverupWall(x::Integer, y::Integer,
width::Integer=8, height::Integer=8, tiletype::String="3", alpha::Number=1.0,
Depth::Integer=-13000, flag::String="", inverted::Bool=false, instant::Bool=true, RenderPlayerOver::Bool=false)

const placements = Ahorn.PlacementDict(
    "Coverup Wall (Customizable) (Viv's Helper)" => Ahorn.EntityPlacement(
        CustomCoverupWall,
        "rectangle",
        Dict{String, Any}("blendIn" => true),
        Ahorn.tileEntityFinalizer
    ),
)
Ahorn.editingOptions(entity::CustomCoverupWall) = Dict{String, Any}(
    "tiletype" => Ahorn.tiletypeEditingOptions(),
    "Depth" => VivHelper.Depths,
    "instant" => Dict{String, Bool}("Instant" => true, "Gradual" => false)
)
Ahorn.minimumSize(entity::CustomCoverupWall) = 8, 8
Ahorn.resizable(entity::CustomCoverupWall) = true, true
Ahorn.selection(entity::CustomCoverupWall) = Ahorn.getEntityRectangle(entity)
Ahorn.renderAbs(ctx::Ahorn.Cairo.CairoContext, entity::CustomCoverupWall, room::Maple.Room) = Ahorn.drawTileEntity(ctx, room, entity, alpha=0.7 * clamp(get(entity.data, "alpha", 1.0), 0.4, 1.0))
end

File: Loenn/entities/CustomCoverupWall.lua
local fakeTilesHelper = require('helpers.fake_tiles')

local coverupWall = {}

coverupWall.name = "VivHelper/CustomCoverupWall"
coverupWall.depth = function(room,entity) return entity.Depth or -13000 end
coverupWall.placements = {
    name = "main",
    data = {
        tiletype = "3",
        width = 8,
        height = 8,
        alpha = 1.0,
        Depth = -13000,
        instant = true,
        flag = "",
        inverted = false
    },
    fieldInformation = {instant = {options = {{"Instant", true}, {"Gradual", false}}}}
}

coverupWall.sprite = function(room,entity) return fakeTilesHelper.getEntitySpriteFunction("tiletype", "blendIn", "tilesFg", {entity.alpha, entity.alpha, entity.alpha, entity.alpha * 0.7})(room,entity) end
coverupWall.fieldInformation = fakeTilesHelper.getFieldInformation("tiletype")

return coverupWall